cloud/google/serviceaccount.go

```diff
@@ -26,68 +23,142 @@ import (
 )
 
 const (
-	ServiceAccountPrefix    = "k8s-machine-controller-"
-	ServiceAccount          = "service-account"
-	MachineControllerSecret = "machine-controller-credential"
+	MasterNodeServiceAccountPrefix        = "k8s-master"
+	WorkerNodeServiceAccountPrefix        = "k8s-worker"
+	MachineControllerServiceAccountPrefix = "k8s-machine-controller"
+	MachineControllerSecret               = "machine-controller-credential"
+	ClusterAnnotationPrefix               = "service-account-"
+)
+
+var (
+	MasterNodeRoles = []string{
+		"compute.instanceAdmin",
+		"compute.networkAdmin",
+		"compute.securityAdmin",
+		"compute.viewer",
+		"iam.serviceAccountUser",
+		"storage.admin",
+		"storage.objectViewer",
+	}
+	WorkerNodeRoles = []string{
+		"compute.instanceAdmin",
+		"compute.networkAdmin",
+		"compute.securityAdmin",
+	}
+	MachineControllerRoles = []string{
+		"compute.instanceAdmin.v1",
+		"iam.serviceAccountActor",
+	}
 )
 
+// Returns the email address of the service account that should be used
```

Comment:
Would this be better stored in structured provider status for the cluster? Annotations are fine for now.

Reply:
https://github.com/kubernetes-sigs/cluster-api/blob/master/pkg/apis/cluster/v1alpha1/cluster_types.go#L99 you can create one for GCE.

Reply:
I want to create a cluster actuator and add onto the current cluster controller in the near future. Do you think it would be ok to wait until then to create this? I think I'll understand what all belongs in the cluster status better by that point.